modin/pandas/indexing.py

```diff
@@ -810,16 +908,24 @@ def _compute_enlarge_labels(self, locator, base_index):
         # base_index_type can be pd.Index or pd.DatetimeIndex
         # depending on user input and pandas behavior
         # See issue #2264
-        base_index_type = type(base_index)
-        locator_as_index = base_index_type(locator)
+        base_as_index = pandas.Index(list(base_index))
+        locator_as_index = pandas.Index(list(locator))
```

Comment:
why? this seems to be breaking the index type...

Reply:
I had issues with this when the Index type of the base_index was incompatible with the locator Index type.

For example, when `base_index` has type `pandas.RangeIndex` and `locator` has type `pandas.Int64Index`, the original code errors with `TypeError: Value needs to be a scalar value, was type Int64Index`.

Casting to list extracts all of the actual values and then casts it back to `pandas.Index` so that we can find the intersection of the two indexes.

Reply:
okay, I see that it isn't returned outside, should be fine then